dev-infrastructure/templates/aks-development.bicep

```diff
@@ -123,6 +90,66 @@ resource aks_pod_nsg 'Microsoft.Network/networkSecurityGroups@2023-09-01' = {
   location: location
 }
 
+resource aks_keyvault 'Microsoft.KeyVault/vaults@2023-07-01' = {
+  location: location
+  name: take('${resourceGroup().name}-svc', 24)
```

Comment:
Might need to re-think this approach. On test apply, i got a failure because it cut off on a hyphen

>The vault name 'aro-hcp-private-anatale-' is invalid. A vault's name must be between 3-24 alphanumeric characters. The name must begin with a letter, end with a letter or digit, and not contain consecutive hyphens

Reply:
since its unlikely user ID will be very large (i think most are cut off to 8 characters), maybe starting with user name instead of RG would work. May not help when folks have more than one RG, since i think `uniqueString` will provide same results when providing the same value for hashing but seems like thats required in order to clean up. 
```
take('${currentUserId}-svc-${uniqueString(currentUserId)}', 24)
```

https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-functions-string#uniquestring

Tried playing with using `newGuid()` but linter complains, seems like repeatable name is a good idea
> Resource identifiers should be reproducible outside of their initial deployment context. Resource aks_keyvault's 'name' identifier is potentially nondeterministic due to its use of the 'newGuid' function

The `svc` part can later become somthing like `cluster-type` for svc, mc, etc

Reply:
Updated to generate a valid name in all cases